Source/VCGeneration/Check.cs

```diff
@@ -410,8 +392,12 @@ public void BeginCheck(string descriptiveName, VCExpr vc, ProverInterface.ErrorH
       this.handler = handler;
 
       thmProver.Reset(gen);
-      SetTimeout();
-      SetRlimit();
+      SetTimeout(timeout);
+      SetRlimit(rlimit);
+      if (randomSeed.HasValue)
+      {
+        SetRandomSeed(randomSeed.Value);
+      }
```

Comment:
In the else case, shouldn't the random seed be set to the default value 0? Because otherwise a custom seed of one implementation would leak to a subsequent implementation that does not declare a custom seed. Basically this means that a "set seed" command should be sent to the solver for every VC (as it is the case for timeout and rlimit).
If yes, I would move the if-then-else statement into `SetRandomSeed`.

Reply:
I just tested, and there is actually a `(set-option :smt.random_seed ...)` command issued for every VC, "resetting" to `0` if nothing is specified on the implementation. But judging by the way the `RandomSeed` property of `NamedDeclaration` is implemented, this is by accident.
`CheckIntAttribute` has the somewhat unintuitive behavior of returning false only when there was a "parsing error" (i.e., when an attribute with the right name is found, but it is not an integer expression). However, it returns true and leaves the ref parameter unchanged when no attribute with the right name is found.
Shouldn't `RandomSeed` be non-nullable and implemented similarly as `TimeLimit` and `ResourceLimit`?

Reply:
If we were to set random seed before verifying each implementation to 0, that would change the behavior currently implemented in Boogie?  Are you saying that would be a change for the better?

Reply:
I suppose it would add more determinism.  We have often seen how individual implementations may be verified alone using /proc: but not when verified together with other implementations (and vice versa).  By ensuring that the random seed is set to 0 (which also happens to be Z3's default), we will eliminate this source of nondeterminism.

Reply:
After some thinking, I concluded that resetting the random seed to 0 before pushing each verification condition is indeed a step in the right direction and should help us reduce the butterfly effect in Boogie usage.  The principle here is that provability of a stable verification condition should not depend on the random seed but if it does it should be explicitly admitted via an annotation.